--- ../busybox-1.23.2.orig/libbb/missing_syscalls.c	2015-03-23 06:06:55.000000000 +0300
+++ ./libbb/missing_syscalls.c	2015-09-27 16:54:13.080524513 +0300
@@ -39,4 +39,15 @@
 {
 	return syscall(__NR_pivot_root, new_root, put_old);
 }
+
+int swapoff(const char *path)
+{
+  return syscall(__NR_swapoff, path);
+}
+
+int swapon(const char *path, int swapflags)
+{
+  return syscall(__NR_swapon, path, swapflags);
+}
+
 #endif
