diff -urN ../busybox-1.25.1.orig/include/libbb.h ./include/libbb.h
--- ../busybox-1.25.1.orig/include/libbb.h	2016-10-07 17:47:47.000000000 +0300
+++ ./include/libbb.h	2016-12-23 11:00:20.567033487 +0300
@@ -284,9 +284,9 @@
 /* Users report bionic to use 32-bit off_t even if LARGEFILE support is requested.
  * We misdetected that. Don't let it build:
  */
-struct BUG_off_t_size_is_misdetected {
-	char BUG_off_t_size_is_misdetected[sizeof(off_t) == sizeof(uoff_t) ? 1 : -1];
-};
+//struct BUG_off_t_size_is_misdetected {
+//	char BUG_off_t_size_is_misdetected[sizeof(off_t) == sizeof(uoff_t) ? 1 : -1];
+//};
 
 /* Some useful definitions */
 #undef FALSE
